Kitodo/src/main/java/org/kitodo/production/model/bibliography/course/Course.java

```diff
@@ -382,6 +387,12 @@ private void processRecoveredMetadata(List<RecoveredMetadata> recoveredMetadata)
             }
             CountableMetadata metadata = new CountableMetadata(foundBlock, Pair.of(metaDatum.getDate(), foundIssue));
             metadata.setMetadataType(metaDatum.getMetadataType());
+            ProcessSimpleMetadata processSimpleMetadata = possibleProcessDetails.get(metaDatum.getMetadataType());
```

Comment:
Potential null pointer exception if `possibleProcessDetails` ist set to `null` like in the test execution (see CourseTest.java line 34). The test code execution did not cover this peace of code.

Reply:
The field is not nullable when a course is loaded with metadata, and it is the result of a stream collect operation, which can return empty, but never returns `null`. In the test situation, it works with `null` because there is no known in the test course. But, I will input `emptySet()` in the test for improvement, it is definitely cleaner.